packages/trie/test/trie/trie.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { Level } from 'level'
+import { MemoryLevel } from 'memory-level'
+import { tmpdir } from 'os'
+import * as tape from 'tape'
+import { LevelDB, ROOT_DB_KEY, Trie } from '../../src'
+
+tape('Trie', function (t) {
+  t.test(
+    'creates an instance via the static constructor `make` function and defaults to `true` with a database',
+    async function (st) {
+      st.true(((await Trie.create({ db: new LevelDB(new Level(tmpdir())) })) as any)._persistRoot)
+    }
+  )
+
+  t.test(
+    'creates an instance via the static constructor `make` function and respects the `persistRoot` option with a database',
+    async function (st) {
+      st.false(
+        ((await Trie.create({ db: new LevelDB(new Level(tmpdir())), persistRoot: false })) as any)
+          ._persistRoot
+      )
+    }
+  )
+
+  t.test(
+    'creates an instance via the static constructor `make` function and defaults to `false` without a database',
+    async function (st) {
+      st.false(((await Trie.create()) as any)._persistRoot)
+    }
+  )
+
+  t.test('persist the root if the `persistRoot` option is not `false`', async function (st) {
+    const trie = await Trie.create({ db: new LevelDB(new MemoryLevel()) })
+
+    st.equal(await trie.db.get(ROOT_DB_KEY), null)
```

Comment:
Should this not return the roothash of the empty trie instead of null?

Reply:
Elsewhere you have this:
```
st.notEqual(await copy.db.get(ROOT_DB_KEY), null)
```
Did you mean to use `notEqual` here?

Reply:
> Should this not return the roothash of the empty trie instead of null?

This is just retrieving the value stored under the special `__root__` key in the DB.  The roothash of the empty trie is returned if you call `trie.root` (which is different).  I guess we could programmatically set the `__root__` key to the hash of the empty trie if a DB is provided but not sure what value that adds.

Reply:
> @acolytec3 @jochem-brouwer I guess we could call `persistRoot()` in `create` if no root has been stored yet? Don't really see the benefit of that since the value will always be the same for an empty trie but could be an option.

If there is an `opts.root`, I think it makes sense. So I would just add a call to persist root in the constructor where you set root of that opt is provided.

Reply:
Sorry, was hard to do on my phone.  LIke [this](https://github.com/ethereumjs/ethereumjs-monorepo/pull/2071/files#r931122810)

Reply:
Given my other suggestion won't work (can't do async ops inside sync constructor), take the above route except write to the DB in the else branch instead of doing that boolean check.  The DB will be still keep the key when you instantiate the trie so no need to have a separate boolean to check what we already know to be true (i.e. `shouldPersistRoot`).

Reply:
LGTM